### PR TITLE
Promote Patient Finder to top-level menu

### PR DESCRIPTION
**Pull request by robertdown**
Move the Patient Finder menu item from underneath Patient/Client to the top level as the second menu item.

### CODE REVIEW

**Comment by sjpadgett** [comment]
The top level menu is already over crowded. With Brady adding back the About page and several features turned on then we'll have to start collapsing menu on laptops views or col-xl. I recm turning on all possible globals that appear in top level and move some if needed.
Also user can turn on patient finder as one of their two default tabs if finder use is needed to be prominent.
anyway, this is 1280px width. I don't think we break till 1200 for some reason. That needs to be fixed. Also, there is too much padding used on top menu items. We currently need to collapse at 1280px.

image

**Comment by bradymiller** [comment]
Also, a quick way to get to patient finder is to click the search box

**Comment by robertdown** [comment]
The patient finder is probably a high-traffic screen, I think we should really consider making it top-level. It's arguably as important in a clinical workflow as the calendar, flow board, and recall board (if not more important than the recall board)

**Comment by robertdown** [comment]
Also, agree there is too much padding. I'll work on reducing the padding (and maybe nudging down the font-size of the menu a bit)

**Comment by robertdown** [comment]
Possible idea would also be to have this menu customizable by user and by role (more so than it already is).

**Comment by sjpadgett** [comment]
> The patient finder is probably a high-traffic screen,

Oh I agree Robert! I'm just pointing out something needs to be done because either we fix the collapse breakpoint to break at 1280px or some items have to go from top level. There are a lot of folks working with 1280x800 views and it becomes a poor UX when menu is collapsed as there default view. I don't know what the tradeoff should be.

**Comment by sjpadgett** [comment]
> (and maybe nudging down the font-size of the menu a bit)

is already too small on light theme.

**Comment by bradymiller** [comment]
Maybe change:
`Flow Board` to `Flow`
`Recall Board` to `Recall`
`Patient Finder` to `Finder`
`Patient\Client` to `Patient`
`Administration` to `Admin`
`Miscellaneous` to `Misc`

**Comment by robertdown** [comment]
I think all of those are good options (and low-hanging fruit). The only bummer is we lose the translation constants.

**Comment by adunsulag** [comment]
Can we use the CSS text-overflow: ellipsis property on the menu and give the menu items a constrained width?  If we make sure the full title is in the aria and title properties this may be good enough as they can still see the titles on hover?  That way you don't lose translations.

**Comment by robertdown** [comment]
Do we really have a large user base using 1200x768? Wouldn't 1280x768 be a more likely resolution? Or even 1366x768? We can also look at hiding the menu logo on smaller screens

**Comment by sjpadgett** [comment]
> Wouldn't 1280x768

That's what I was pointing out. Currently we collapse at 1200px should be 1280px.

**Comment by robertdown** [comment]
Ok, I reduced the X-padding down some (Probably can't reduce it anymore and maintain readability), dropped "Board" from the menu items and here is a screenshot on a 1200x768 view

image

**Comment by sjpadgett** [comment]
@bradymiller Did you already move the About page? Did you want that on top level?
Also Robert you may want to move newcrop or turn on to check menu with it on. I don't know of any other features that get on top level if feature is turned on besides Therapy Groups.

**Comment by robertdown** [comment]
Think we could put the About link under the User menu? That way it is still reachable by everyone but is not top level

**Comment by bradymiller** [comment]
Under User menu would work also since will always show up as option (will be clear where it is after User uses OpenEMR a couple times; would prob need to call it `About OpenEMR` so User does not think it is about themselves :) ) when user logs out (goal is to make it as accessible as possible since it has a donate button).

**Comment by bradymiller** [comment]
also, looks like all tests are failing; guessing a simple issue in menu change.

**Comment by sjpadgett** [comment]
Maybe could also stick it under provider pull for logout and password reset. It'd get seen there a lot...

**Comment by robertdown** [comment]
**IMPORTANT** - I committed on this branch branch and pushed it before realizing my mistake; I was on the wrong branch. A commit which was intended for https://github.com/openemr/openemr/pull/4955 was accidentally committed to this PR.

I did a local reset and did a force push to return this PR to the proper state.

**Comment by robertdown** [comment]
Dropped user's name from the menu. Here is the new User menu option.

image

File: interface/main/tabs/templates/user_data_template.php [hunk 1]
```diff
@@ -18,17 +18,24 @@
...
+                        <li class="menuLabel">
+                            <span class="font-weight-bold"><span data-bind="text:fname"></span> <span data-bind="text:lname"></span></span>
+                        </li>
+                        <li class="menuLabel" data-bind="click: editSettings"><i class="fa fa-fw pr-2 text-muted fa-cog"></i>&nbsp;<?php echo xlt("Settings");?></li>
+                        <li class="menuLabel" data-bind="click: changePassword"><i class="fa fa-fw pr-2 text-muted fa-lock"></i>&nbsp;<?php echo xlt("Change Password");?></li>
+                        <li class="menuLabel" data-bind="click: changeMFA"><i class="fa fa-fw pr-2 text-muted fa-key"></i>&nbsp;<?php echo xlt("MFA Management");?></li>
+                        <div class="dropdown-divider"></div>
+                        <li class="menuLabel" data-bind="click: function() {navigateTab('/interface/main/about_page.php', 'About');activateTabByName('msc');}"><i class="fa fa-fw pr-2 text-muted fa-info"></i>&nbsp;<?php echo xlt("About");?> <?php echo $GLOBALS['openemr_name']; ?></li>
```
Comment (bradymiller) [inline interface/main/tabs/templates/user_data_template.php:35]: text() on $GLOBALS['openemr_name']

File: interface/main/tabs/menu/menus/standard.json [hunk 2]
```diff
@@ -16,7 +16,19 @@
     ]
```
Comment (bradymiller) [inline interface/main/tabs/menu/menus/standard.json:16]: Would remove the About menu item in this menu (it is in Admin)

File: interface/main/tabs/menu/menus/standard.json [hunk 3]
```diff
@@ -16,7 +16,19 @@
...
   },
   {
```
Comment (bradymiller) [inline interface/main/tabs/menu/menus/standard.json:18]: When this file is ready, also should copy it to sites/default/documents/custom_menus/Custom.json
(this is basically a copy of standard.json, which folks can use as a starting point to create their own custom menus)

**Comment by bradymiller** [comment]
screen shot of user menu is beautiful!

2DV-2

**Comment by bradymiller** [comment]
Looks like all function e2e tests are failing at `Check add user` test. 
should be able to fix this by changing `Administration` to `Admin` at line 13 of tests/Tests/E2e/Pages/MainPage.php

**Pull request merged**